dlls/client.cpp

```diff
@@ -546,11 +546,21 @@ void ClientCommand(edict_t* pEntity)
 	}
 	else if (FStrEq(pcmd, "use"))
 	{
-		player->SelectItem((char*)CMD_ARGV(1));
+		auto weapon = player->GetNamedPlayerItem((char*)CMD_ARGV(1));
+		
+		if (weapon != nullptr)
+		{
+			player->SelectItem(weapon);
+		}
 	}
 	else if (((pstr = strstr(pcmd, "weapon_")) != NULL) && (pstr == pcmd))
 	{
-		player->SelectItem(pcmd);
+		auto weapon = player->GetNamedPlayerItem(pcmd);
+
+		if (weapon != nullptr)
```

Comment:
It would be better to have a `SelectItem` overload that takes a classname and do what the original method did, then you don't need these changes here. People who merge changes into their projects won't have to make extra changes as well.

Reply:
Good idea. If I update my branch, would you recommend amending the existing commit or creating a new one?

Reply:
Amend the commit. No need to add multiple commits to a pull request.